### PR TITLE
3d upload

### PR DESCRIPTION
**Pull request by vince0656**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -59,18 +59,7 @@ const lookupEtherscanAddress = (id) => {
...
-            return 'http://localhost:5000/rendar-token/us-central1/main/api';
-        // case 'rendar.org':
-        // case 'dapp.rendar.org':
-        // case 'rendar-token.web.app':
-        // case 'rendar-token.firebaseapp.com':
-        //     return `https://${window.location.hostname}/api`;
-        default:
-            return 'https://rendar-token.firebaseapp.com/api';
-    }
+    return 'https://rendar-token.firebaseapp.com/api';
```
Comment (vince0656) [inline src/utils.js:62]: @jamesmorgan if you want me to restore the old code let me know

File: src/components/InteractivePlayer.vue [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  <!-- glb editor - https://modelviewer.dev/editor/ -->
+  <!-- reveal="auto"    - If set the model will be revealed as soon as it is done loading and rendering -->
+  <!-- poster           - like a video this is the image behind the loaded 3d object -->
+  <!-- camera-controls  - Enables controls via mouse/touch when in flat view. -->
+  <!-- auto-rotate      - Enables the auto-rotation of the model -->
+  <!-- ar               - Enable the ability to launch AR experiences on supported devices. -->
+  <!-- environment-image- Defines the specific lighting mode, by default we use a simple generic one. -->
+  <!-- interaction-policy-If set to "always-allow", the user can control it even if the viewer is not the focused element on the page-->
+
+  <!-- TODO work out why the card redirect doesnt work ?? -->
```
Comment (jamesmorgan) [inline src/components/InteractivePlayer.vue:12]: chop

File: src/components/InteractivePlayer.vue [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  <!-- :alt="metadata.name" -->
```
Comment (jamesmorgan) [inline src/components/InteractivePlayer.vue:13]: action

File: src/utils.js (hunk 1 above)
Comment (jamesmorgan) [reply src/utils.js:62]: No its fine pointing to live

File: src/components/admin/CreateEdition.vue [hunk 4]
```diff
@@ -288,75 +277,201 @@
...
+                let additionalAttributes = {};
+
+                if (this.imageUpload.is3DAsset) {
+                  topLevelAttributes = {
+                    image: this.coverImageUpload.ipfsImage,
+                    animation_url: this.imageUpload.ipfsImage
+                  }
+
+                  additionalAttributes = {
+                    "asset_type": "model/gltf-binary",
```
Comment (jamesmorgan) [inline src/components/admin/CreateEdition.vue:317]: Is this correct - I assume it should read the file type in some form?

Comment (jamesmorgan) [reply src/components/admin/CreateEdition.vue:317]: Ah ignore me ... I see the if

File: src/components/admin/CreateEdition.vue [hunk 5]
```diff
@@ -288,75 +277,201 @@
...
+                    "asset_size_in_bytes": this.imageUpload.file ? this.imageUpload.file.size : 0,
+                    "cover_image_type": this.coverImageUpload.file ? this.coverImageUpload.file.type : '',
+                    "cover_image_size_in_bytes": this.coverImageUpload.file ? this.coverImageUpload.file.size: 0
+                  }
+                } else {
+                  topLevelAttributes = {
```
Comment (jamesmorgan) [inline src/components/admin/CreateEdition.vue:323]: Can we also add the `asset_type` & `asset_size_in_bytes` when its not 3D as well

File: src/components/admin/CreateEdition.vue (hunk 4 above)
Comment (vince0656) [reply src/components/admin/CreateEdition.vue:317]: At the moment this logic is only triggered when a `.glb` is detected so thought that hard coding it was safe as there's only 1 file type that can trigger. But I can change this if you prefer?

File: src/components/InteractivePlayer.vue (hunk 3 above)
Comment (vince0656) [reply src/components/InteractivePlayer.vue:13]: done

**Review by jamesmorgan — CHANGES_REQUESTED** [review]

File: src/components/InteractivePlayer.vue (hunk 2 above)
Comment (vince0656) [reply src/components/InteractivePlayer.vue:12]: ✂️

File: src/components/admin/CreateEdition.vue (hunk 5 above)
Comment (vince0656) [reply src/components/admin/CreateEdition.vue:323]: your wish is my command 😄

Comment (vince0656) [reply src/components/admin/CreateEdition.vue:323]:
```
{
"name": "d",
"description": "d",
"attributes": {
    "tags": [
        "urban art"
     ],
     "asset_type": "image/png",
     "asset_size_in_bytes": 90719
},
"external_uri": "https://www.rendar.org/",
"image": "https://ipfs.infura.io/ipfs/QmUF3P9NnRqpUR2i8zG7H4MVAYKpe5ZVhV4PFPDYxJKjDC/asset.png"
}
```

**Pull request merged**